src/pages/workspaces/workspace/WorkspaceContainer.js

```diff
@@ -141,109 +140,88 @@ class WorkspaceContainer extends Component {
-        namespace, name, activeTab, showTabBar, workspace, clusters,
-        title: _.isFunction(title) ? title(this.props) : title,
-        breadcrumbs: breadcrumbs(this.props),
-        topBarContent: topBarContent && topBarContent({ workspace, ...this.props }),
-        refresh: async () => {
-          await this.refresh()
-          const child = this.child.current
-          if (child.refresh) {
-            child.refresh()
-          }
-        },
-        refreshClusters: () => this.refreshClusters()
-      }, [
-        workspace && h(WrappedClassComponent, {
-          ref: this.child,
-          workspace, clusters, loadingWorkspace,
-          refreshWorkspace: () => this.refresh(),
-          refreshClusters: () => this.refreshClusters(),
-          ...this.props
-        })
-      ])
-    }
 
-    renderError() {
-      const { workspaceError, errorText } = this.state
-      const groupURL = 'https://software.broadinstitute.org/firecloud/documentation/article?id=9553'
-      const authorizationURL = 'https://software.broadinstitute.org/firecloud/documentation/article?id=9524'
-
-      return div({ style: { padding: '2rem' } }, [
-        workspaceError.status === 404 ?
-          h(Fragment, [
-            h2({}, ['Could not display workspace']),
-            p({},
-              ['You are trying to access a workspace that either does not exist, or you do not have access to it.']),
-            p({}, [
-              'To view an existing workspace, the owner of the workspace must share it with you or with a ',
-              link({ target: '_blank', href: groupURL }, 'Group'), ' of which you are a member. ',
-              'If the workspace is protected under an ', link({ target: '_blank', href: authorizationURL }, 'Authorization Domain'),
-              ', you must be a member of every group within the Authorization Domain.'
-            ]),
-            p({}, [
-              'If you think the workspace exists but you do not have access, please contact the workspace owner.'
-            ]),
-            buttonPrimary({
-              as: 'a',
-              href: Nav.getLink('workspaces')
-            }, ['Return to Workspace List'])
-          ]) :
-          h(Fragment, [
-            h2({}, ['Failed to load workspace']),
-            h(ErrorView, { error: errorText })
-          ])
-      ])
-    }
-
-    componentDidMount() {
-      this.refresh()
-      this.refreshClusters()
-    }
+export const wrapWorkspace = ({ breadcrumbs, activeTab, title, topBarContent, showTabBar = true, queryparams }) => WrappedComponent => {
+  const WrappedClassComponent = toClass(WrappedComponent)
 
-    async refreshClusters() {
-      const { namespace, ajax: { Jupyter } } = this.props
+  const Wrapper = props => {
+    const { namespace, name } = props
+    const child = useRef()
+    const signal = useCancellation()
+    const [accessError, setAccessError] = useState(false)
+    const [workspace, setWorkspace] = useState(undefined)
```

Comment:
Could we get a default for this from the workspace list cache atom from your other PR?

Reply:
Ideally yes, but they don't actually contain the same information.